pagopa/io-app
# feat(Piattaforma Notifiche): [IAMVL-79] Fetch and show PN message details

## Short description
### ⚠️ **This PR depends on #4036** 

This branch implements the detail screen for messages delivered by PN (i.e. `category === "PN"`). Message details are fetched as `ThirdPartyMessage` (using a new `io-backend` endpoint) and stored as `pot` in the `entities.messages.thirdPartyById` section of the store. A third party message has a `details` property (defined as a dictionary) that depends on the third party and must be decoded at runtime by the app. This is done by the `pnMessageFromIdSelector`.

https://user-images.githubusercontent.com/467098/179462946-4ab2229c-406e-4650-929b-eb19fc79e139.mp4

## List of changes proposed in this pull request
- A `MessageLoading` component has been extracted from `MvlLoadingScreen`;
- Add support for new `ThirdPartyMessage` APIs in `io-backend` (actions, store, reducers, saga);
- Add custom models for parsing message details from PN
- Add new `PnStackNavigator` supporting the new message detail screen for PN

## How to test
Use the [IIP-18-add-pn-endpoints](https://github.com/pagopa/io-dev-api-server/tree/IIP-18-add-pn-endpoints) branch of the dev server and add this to your `config.json`:
```
{
  "messages": {
    "pnCount": 1
  },
  "services": {
    "includePn": true
  }
}
```
Set `PN_ENABLED=YES` to `.env` and run the app, then open a message from PN and verify that details are loaded.
